mmdeploy/codebase/mmdet3d/models/centerpoint.py

```diff
@@ -0,0 +1,198 @@
+        bbox_preds (list[torch.Tensor]): Bbox predicts results.
+        dir_scores (list[torch.Tensor]): Dir predicts results.
+        img_metas (list[dict]): Point cloud and image's meta info.
+        img (torch.Tensor): Input image.
+        rescale (Bool): Whether need rescale.
+
+    Returns:
+        list[dict]: Decoded bbox, scores and labels after nms.
+    """
+    rets = []
+    # common_heads = self.task_heads[0].heads
+    batch_size = 1
+    scores_range = [0]
+    bbox_range = [0]
+    dir_range = [0]
+    self.test_cfg = self.test_cfg['pts']
+    for i, task_head in enumerate(self.task_heads):
+        scores_range.append(scores_range[i] + self.num_classes[i])
+        bbox_range.append(bbox_range[i] +
+                          8 if 'vel' in task_head.heads.keys() else 6)
+        dir_range.append(dir_range[i] + 2)
+    for task_id in range(len(self.num_classes)):
+        num_class_with_bg = self.num_classes[task_id]
+
+        batch_heatmap = cls_scores[
+            0][:, scores_range[task_id]:scores_range[task_id + 1],
+               ...].sigmoid()
+
+        batch_reg = bbox_preds[0][:,
+                                  bbox_range[task_id]:bbox_range[task_id] + 2,
+                                  ...]
+        batch_hei = bbox_preds[0][:, bbox_range[task_id] +
+                                  2:bbox_range[task_id] + 3, ...]
+
+        if self.norm_bbox:
+            batch_dim = torch.exp(bbox_preds[0][:, bbox_range[task_id] +
+                                                3:bbox_range[task_id] + 6,
+                                                ...])
+        else:
+            batch_dim = bbox_preds[0][:, bbox_range[task_id] +
+                                      3:bbox_range[task_id] + 6, ...]
+
+        batch_vel = bbox_preds[0][:, bbox_range[task_id] +
+                                  6:bbox_range[task_id + 1], ...]
+
+        batch_rots = dir_scores[0][:,
+                                   dir_range[task_id]:dir_range[task_id + 1],
+                                   ...][:, 0].unsqueeze(1)
+        batch_rotc = dir_scores[0][:,
+                                   dir_range[task_id]:dir_range[task_id + 1],
+                                   ...][:, 1].unsqueeze(1)
+
+        # if 'vel' in preds_dict[0]:
+        #     batch_vel = preds_dict[0]['vel']
+        # else:
+        #     batch_vel = None
+
+        temp = self.bbox_coder.decode(
+            batch_heatmap,
+            batch_rots,
+            batch_rotc,
+            batch_hei,
+            batch_dim,
+            batch_vel,
+            reg=batch_reg,
+            task_id=task_id)
+        assert self.test_cfg['nms_type'] in ['circle', 'rotate']
+        batch_reg_preds = [box['bboxes'] for box in temp]
+        batch_cls_preds = [box['scores'] for box in temp]
+        batch_cls_labels = [box['labels'] for box in temp]
+        if self.test_cfg['nms_type'] == 'circle':
+            ret_task = []
+            for i in range(batch_size):
```

Comment:
do we support batch inference for centerpoint? if we loop through batch dim, batch size would be static.